packages/gatsby/cache-dir/page-renderer.js

```diff
@@ -11,6 +12,15 @@ class PageRenderer extends React.Component {
       pathContext: this.props.pageContext,
     }
 
+    if (this.props.pageContext.matchPath) {
+      const { params } = match(
+        this.props.pageContext.matchPath,
+        this.props.location.pathname
+      )
+      props.pageContext.params = params
+      props.pathContext.params = params
```

Comment:
2x? To path and page?

Reply:
I guess it's for compatibility purpose. Cuz right above there is `pathContext: this.props.pageContext,`

Reply:
Yeah, we changed it between v1 and v2 and kept `pathContext` (alias really) for backward compatibility.

That said, I feel like this shouldn't be in neither of them and maybe this should be top level prop? I need to check how @reach/router does this - because I feel like <Router> should automatically give us this props and we just might drop those on way from <Router> to user's page component (I don't think we should need to match again, as we already matched before that)

Reply:
So reach/router, given `invoices/:invoiceId` matchPath will use following props:
```
{
  "path": "invoices/:invoiceId",
  "invoiceId": "123",
  "uri": "/invoices/123",
  "location": {
    "replace": "fn()",
    "href": "https://k9lzn45065.csb.app/invoices/123",
    "ancestorOrigins": "DOMStringList(…)",
    "origin": "https://k9lzn45065.csb.app",
    "protocol": "https:",
    "host": "k9lzn45065.csb.app",
    "hostname": "k9lzn45065.csb.app",
    "port": "",
    "pathname": "/invoices/123",
    "search": "",
    "hash": "",
    "assign": "assign()",
    "reload": "reload()",
    "toString": "toString()",
    "state": {
      "key": "1567638513581"
    },
    "key": "1567638513581"
  },
  "navigate": "navigate()"
}
```
so param is being inserted just as top level prop - it sounds like good idea to preserve this behaviour - what do you think?

Reply:
Thanks for reviewing. That looks good too. However, there is a rare edge case where the param key is the same as one of the props e.g. `/uris/:uri` or `/paths/:path`. But I guess that's not a big concern.
cc @slorber

Reply:
Anw looks like @reach/router alr does this. From Page component, we can simply access `props[paramKey]` to get our param.
One important caveat is @reach/router will return `null` when we access index path e.g. `/user` for matchPath `/user/:userId`.
Guess we need to update the doc then.

Reply:
Yeah that would be fine if we just forwarded the gatsby match call result to the page instead of rematching.

Reply:
And what I meant is we cal alr do it currently. Just use props[paramKey] from page components.